### ISSUE TITLE
Combining multiple stopping conditions

### ISSUE THREAD

**Issue by apaleyes** [amzn/emukit#273]
At the moment (once #272 is merged) we support combinations of stopping conditions with an "or" operation. That is, given a list of conditions the loop will continue until one of them fires, and then the loop terminates.

It would be nice to have more flexible support here, so that users could combine conditions in various ways. Example use case: i'd like to run optimization until it converges, but make at least 10 steps, and to prevent it from running indefinitely i'd like to stop after 50 steps. Our API should allow for that:
```
(cond1 AND cond2) OR cond3
```

we can consider overloading python's `and` and `or` for that.

**Comment by apaleyes** [comment]
implemented
